### PR TITLE
Move SITL Compass to standard sensor backend model.

### PR DESCRIPTION
**Pull request by mayanez**
@tridge As we discussed, this makes compass like `AP_Baro_SITL`.

### CODE REVIEW

File: libraries/AP_Compass/AP_Compass_SITL.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    if (best_time_delta < 1000) { // only output stored state if < 1 sec retrieval error
+        new_mag_data = buffer[best_index].data;
+    }
+
+    new_mag_data -= _sitl->mag_ofs.get();
+
+    for (uint8_t i=0; i<SITL_NUM_COMPASSES; i++) {
+        rotate_field(new_mag_data, _compass_instance[i]);
+        publish_raw_field(new_mag_data, AP_HAL::micros(), _compass_instance[i]);
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass_SITL.cpp]: looks like we're missing a call to correct_field() ? We need that to take offsets into account

Comment (mayanez) [inline libraries/AP_Compass/AP_Compass_SITL.cpp]: You're right, I missed that for some reason.

**Comment by mayanez** [comment]
Just updated it.

File: libraries/AP_Compass/AP_Compass_SITL.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    }
+    if (best_time_delta < 1000) { // only output stored state if < 1 sec retrieval error
+        new_mag_data = buffer[best_index].data;
+    }
+
+    new_mag_data -= _sitl->mag_ofs.get();
+
+    for (uint8_t i=0; i<SITL_NUM_COMPASSES; i++) {
+        correct_field(new_mag_data, _compass_instance[i]);
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass_SITL.cpp]: rotate_field needs to be done first, then publish that raw field, then call correct_field
The compass calibrator reliies on the ordering. It needs the uncorrected field for calibration, but it needs to be rotated

Comment (mayanez) [inline libraries/AP_Compass/AP_Compass_SITL.cpp]: Ahh, good to know. I will update it

**Comment by tridge** [comment]
merged, thanks!

**Pull request merged**